Tensile/TensileCreateLibrary.py

```diff
@@ -1165,22 +1306,68 @@ def TensileCreateLibrary():
       "Tools.cpp",
       "Tools.h" ]
   else:
-    libraryStaticFiles = [
+    staticFiles = [
       "TensileTypes.h",
       "tensile_bfloat16.h",
       "KernelHeader.h" ]
 
-  # write cmake
-  clientName = "LibraryClient"
-  writeCMake(outputPath, solutions, kernels, libraryStaticFiles, clientName )
+  # Build a list of files to be expected
+  (solutionFiles,
+   sourceKernelFiles,
+   asmKernelFiles,
+   sourceLibFiles,
+   asmLibFiles) = buildObjectFileNames(solutionWriter, kernelWriterSource, \
+    kernelWriterAssembly, solutions, kernels, kernelsBetaOnly)
+
+  (solutionPaths,
+   sourceKernelPaths,
+   asmKernelPaths,
+   sourceLibPaths,
+   asmLibPaths) = buildObjectFilePaths(outputPath, solutionFiles, sourceKernelFiles, \
+    asmKernelFiles, sourceLibFiles, asmLibFiles)
+
+  if globalParameters["GenerateManifestAndExit"] == True:
 
-  # write solutions and kernels
+    # Generate manifest file
+    libraryPath = os.path.join(outputPath, "library")
+    ensurePath(libraryPath)
+    generatedFile = open(os.path.join(libraryPath, "TensileManifest.txt"), "w")
+
+    libraryFilename = "TensileLibrary.yaml" if globalParameters["LibraryFormat"] == "yaml" else "TensileLibrary.dat"
+
+    # Manifest file contains YAML file, output library paths and cpp source for embedding.
+    for filePath in [os.path.join(libraryPath, libraryFilename)] + sourceLibPaths + asmLibPaths:
+      generatedFile.write("%s\n" %(filePath) )
+    generatedFile.close()
+
+    return
+
+  # generate cmake for the source kernels
+  writeCMake(outputPath, solutionFiles, sourceKernelFiles, staticFiles)
+
+  # Make sure to copy the library static files.
+  for fileName in staticFiles:
+    shutil.copy( os.path.join(globalParameters["SourcePath"], fileName), \
+      outputPath )
+
+    # write solutions and kernels
   problemTypes = list(logicData.keys())
+
   codeObjectFiles = writeSolutionsAndKernels(outputPath, CxxCompiler, problemTypes, solutions,
                                              kernels, kernelsBetaOnly,
                                              solutionWriter,
                                              kernelWriterSource, kernelWriterAssembly)
 
+  sanityCheck0 = set(codeObjectFiles) - set(sourceLibPaths + asmLibPaths)
+  sanityCheck1 = set(sourceLibPaths + asmLibPaths) - set(codeObjectFiles)
+
+  if len(sanityCheck0) > 0:
+    print1("Warning: Manifest is missing library files: {}".format(sanityCheck0))
+
+  if len(sanityCheck1) > 0:
+    print1("Warning: Manifest has unexpected library files: {}".format(sanityCheck1))
```

Comment:
What are the situations where this could happen? Should this be an assert? Should it throw an error?